### PR TITLE
update defaults, introduce multi OS support

### PR DESCRIPTION
**Pull request by ymajik**
rewritten the role to include more variables, OS support (Ubuntu, Fedora); add automated tests for those OS via travis ci.

### CODE REVIEW

File: templates/redshift.conf.j2 [hunk 1]
```diff
@@ -17,7 +25,11 @@ transition=1
 ;brightness-night=0.4
 ; Set the screen gamma (for all colors, or each color channel
 ; individually)
+{% if redshift.temp is defined %}
```
Comment (gvillalta99) [inline templates/redshift.conf.j2]: Probably `redshift.gamma` here

File: templates/redshift.conf.j2 [hunk 2]
```diff
@@ -27,7 +39,11 @@ gamma=0.8
 ; Set the location-provider: 'geoclue', 'geoclue2', 'manual'
 ; type 'redshift -l list' to see possible values.
 ; The location provider settings are in a different section.
+{% if redshift.location_providertemp is defined %}
```
Comment (gvillalta99) [inline templates/redshift.conf.j2]: Misspelled `redshift.location_provider`

File: tests/test.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+- hosts: all
+  #remote_user: root
+
+  roles:
+    - role_under_test
```
Comment (gvillalta99) [inline tests/test.yml]: Does this works?
Shouldn't it be `ansible-redshift` or the name of this role?

**Comment by gvillalta99** [comment]
Great @ymajik, thank you!
I've made some comments on your code, most of them about misspellings and the last one about the test, because I've never tested ansible roles using Docker + Travis.

I'm waiting your feedback.

File: templates/redshift.conf.j2 (hunk 1 above)
Comment (ymajik) [inline templates/redshift.conf.j2]: indeed, fixed it, my bad

File: templates/redshift.conf.j2 (hunk 2 above)
Comment (ymajik) [inline templates/redshift.conf.j2]: indeed, fixed it, my bad

File: tests/test.yml (hunk 3 above)
Comment (ymajik) [inline tests/test.yml]: yes, this works, have a look  at geerlingguy his ansible travis test files. You can replace all the role_under_test occurences with the right rolename, or use role_under_test, to easily reuse the .travis.yml file.

**Comment by ymajik** [comment]
@gvillalta99 thanks for the comments, I fixed the misspellings. I also just started testing ansible roles with Travis and Docker, it has a lot of potential. On https://github.com/geerlingguy/ his repos you can find a lot of ansible roles with good Travis examples.

**Comment by ymajik** [comment]
I fixed some more things and included a dockerfile for debian jessie

**Comment by gvillalta99** [comment]
Thank you @ymajik, those changes are amazing :star2:!
And I will take a look at geerlingguy's repos.

**Pull request merged**